src/v/s3/tests/s3_client_test.cc

```diff
@@ -496,28 +520,9 @@ SEASTAR_TEST_CASE(test_client_pool_reconnect) {
         std::vector<ss::future<bool>> fut;
         for (size_t i = 0; i < 20; i++) {
             auto f = pool->acquire().then(
-              [_server = server](
-                s3::client_pool::client_lease lease) -> ss::future<bool> {
-                  auto server = _server;
-                  co_await ss::sleep(100ms);
-                  auto client = lease.client;
-                  iobuf payload;
-                  auto payload_stream = make_iobuf_ref_output_stream(payload);
-                  try {
-                      auto http_response = co_await client->get_object(
-                        s3::bucket_name("test-bucket"),
-                        s3::object_key("test"),
-                        100ms);
-                      auto input_stream = http_response->as_input_stream();
-                      co_await ss::copy(input_stream, payload_stream);
-                      iobuf_parser p(std::move(payload));
-                      auto actual_payload = p.read_string(p.bytes_left());
-                      BOOST_REQUIRE_EQUAL(actual_payload, expected_payload);
-                      client->shutdown();
-                  } catch (const ss::abort_requested_exception&) {
-                      co_return false;
-                  }
-                  co_return true;
+              [server = server](s3::client_pool::client_lease lease) {
```

Comment:
nit
```suggestion
              [server](s3::client_pool::client_lease lease) {
```

Reply:
server in the parent scope is not a variable and can't be captured.

Reply:
Silly clang!